src/fileio/dataformat.cxx

```diff
@@ -79,8 +79,8 @@ void DataFormat::readFieldAttributes(const std::string& name, FieldPerp& f) {
   // Note: don't use DataFormat::mesh variable, because it may be null if the DataFormat
   // is part of a GridFromFile, which is created before the Mesh.
   if (getAttribute(name, "yindex_global", yindex_global)) {
-    f.setIndex(f.getMesh()->YLOCAL(yindex_global));
+    f.setIndex(f.getMesh()->getLocalYIndexNoBoundaries(yindex_global));
   } else {
-    f.setIndex(f.getMesh()->YLOCAL(0));
+    f.setIndex(f.getMesh()->getLocalYIndexNoBoundaries(0));
```

Comment:
I think it makes sense to keep `getLocalYIndexNoBoundaries(0)` here, as that sets a default value on a grid cell rather than (possibly) in a boundary cell.

Reply:
I'm not sure about this, might be better to just fail?

Reply:
I wanted to not fail so we can successfully read old restart files, which might (I think? haven't checked the actual history...) exist with FieldPerps but without `yindex_global` attributes.

Reply:
That sounds reasonable, I'm just not sure how you'd deal with such a field? You'd need to set the index via some other means